crates/testing/tests/taxman.rs

```diff
@@ -107,48 +107,48 @@ mod taxman {
 // The tx should fail at `withhold_fee` stage.
 // No state change should be committed.
 #[test_case(
-    10,
-    1,
+    10_u128,
+    1_u128,
     100_000,
-    0,
-    10,
-    0,
-    Some("subtraction overflow: 10 - 25000 < u128::MIN");
+    0_u128,
+    10_u128,
+    0_u128,
+    Some("subtraction overflow: 10 - 25000 < bnum::buint::BUint<4>::MIN");
     "error while withholding fee"
 )]
 // Case 2. Sender has enough balance to cover gas fee, but not enough for the
 // transfer.
 // The tx should pass `withhold_fee`, but fail at processing messages.
 // The fee should be deducted from the sender's account, but the transfer reverted.
 #[test_case(
-    30_000,
-    99_999,
+    30_000_u128,
+    99_999_u128,
     100_000,
-    6250,  // = 100,000 / 4 * 0.25
-    23750, // = 30,000 - (100,000 / 4 * 0.25)
-    0,
-    Some("subtraction overflow: 5000 - 99999 < u128::MIN");
+    6250_u128,  // = 100,000 / 4 * 0.25
+    23750_u128, // = 30,000 - (100,000 / 4 * 0.25)
+    0_u128,
+    Some("subtraction overflow: 5000 - 99999 < bnum::buint::BUint<4>::MIN");
     "error while processing messages"
 )]
 // Case 3. Sender has enough balance to cover both gas fee and the transfer.
 // State changes from both gas fee and transfer should be affected.
 #[test_case(
-    30_000,
-    123,
+    30_000_u128,
+    123_u128,
     100_000,
-    6250,  // = 100,000 / 4 * 0.25
-    23627, // = 30,000 - (100,000 / 4 * 0.25) - 123
-    123,
+    6250_u128,  // = 100,000 / 4 * 0.25
+    23627_u128, // = 30,000 - (100,000 / 4 * 0.25) - 123
+    123_u128,
     None;
     "successful tx"
 )]
 fn withholding_and_finalizing_fee_works(
-    sender_balance_before: u128,
-    send_amount: u128,
+    sender_balance_before: impl Into<Uint256>,
+    send_amount: impl Into<Uint256>,
     gas_limit: u64,
-    owner_balance_after: u128,
-    sender_balance_after: u128,
-    receiver_balance_after: u128,
+    owner_balance_after: impl Into<Uint256>,
+    sender_balance_after: impl Into<Uint256>,
+    receiver_balance_after: impl Into<Uint256>,
```

Comment:
@Rhaki We use the `where` syntax everywhere in the codebase for trait bounds, for consistency.

https://github.com/left-curve/grug/blob/main/CONTRIBUTING.md#formatting